### PR TITLE
Initialize datum api chart

### PR DESCRIPTION
**Pull request by tylerauerbeck**
This provides (very much) a skeleton chart to deploy the datum API to a Kubernetes cluster. With the current chart and values file you will get the following:
  - datum API
  - local sqlite backed by PVC
  - in memory openFGA

In the current configmap template, configs pretty much mirror our devauth config example. For follow up PR's, we'll want to look at:
  - making configs configurable via values file
  - enabling persistent openfga
  - enabling postgres backed datum
  - enabling remote sqlite (Turso)

### CODE REVIEW

File: charts/datum/templates/configmap-datum.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    DATUM_DB_PRIMARY_DB_SOURCE: "/data/primary/datum.db?mode=memory&_fk=1"
+    DATUM_DB_SECONDARY_DB_SOURCE: "/data/secondary/datum.db?mode=memory&_fk=1"
+    DATUM_AUTHZ_ENABLED: "true"
+    DATUM_AUTHZ_STORE_NAME: "datum"
+    DATUM_AUTHZ_HOST: "datum-openfga:8080"
+    DATUM_AUTHZ_SCHEME: "http"
+    DATUM_AUTHZ_CREATE_NEW_MODEL: "false"
+    DATUM_EMAIL_SEND_GRID_API_KEY: "SG.FakeAPIKey"
+    DATUM_EMAIL_TESTING: "true"
+    DATUM_EMAIL_URL_BASE: "http://localhost:17608/"
```
Comment (golanglemonade) [inline charts/datum/templates/configmap-datum.yaml:28]: The values of these should probably come from the values file, but that can be a follow-up

File: charts/datum/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      securityContext:
+        {{- toYaml .Values.api.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.api.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          args:
+            - serve
+            - --config=/app/config.yaml
```
Comment (golanglemonade) [inline charts/datum/templates/deployment.yaml:41]: Is there a config.yaml setup?

**Review by golanglemonade — APPROVED** [review]

File: charts/datum/templates/configmap-datum.yaml (hunk 1 above)
Comment (golanglemonade) [reply charts/datum/templates/configmap-datum.yaml:28]: oh you put that in your description :)

File: charts/datum/templates/deployment.yaml (hunk 2 above)
Comment (tylerauerbeck) [reply charts/datum/templates/deployment.yaml]: Nah, this was leftovers from when I was fighting with env vars. 🔥

**Review by matoszz — APPROVED** [review]

**Review by golanglemonade — APPROVED** [review]

**Pull request merged**